src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -227,17 +229,20 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
             bool hasJsonBody,
             EndpointMetadataCollection endpointMetadata)
         {
-            var requestMetadata = endpointMetadata.GetOrderedMetadata<IApiRequestMetadataProvider>();
-            var declaredContentTypes = DefaultApiDescriptionProvider.GetDeclaredContentTypes(requestMetadata);
+            var requestMetadata = endpointMetadata.GetOrderedMetadata<IAcceptsMetadata>();
```

Comment:
We should pass in the acceptsMetadata above so we don't have to query for it again.

Reply:
Oh wait, I see this is a list

Reply:
This is incorrect. The matcher policy does not look at more than one metadata, this shouldn't either.

Reply:
It's perfectly valid to accept multiple content types per schema though. If you mean we're not deduping content types for a given schema then that's different. The matcher policy should allow any of the content types declared however. If it doesn't [examples like this](https://github.com/DamianEdwards/MinimalApiPlayground/blob/main/src/MinimalApiPlayground/Program.cs#L253) won't work.

Reply:
We should be looking up a single piece of metadata, not all of them

Reply:
@davidfowl  see new changes. I am now just looking at ```var acceptsMetadata = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>();``` and passing the ```acceptsMetadata``` to             ```AddSupportedRequestFormats(apiDescription.SupportedRequestFormats, hasJsonBody, acceptsMetadata);```

Reply:
@davidfowl see new changes. I am now just looking at ```var acceptsMetadata = routeEndpoint.Metadata.GetMetadata<IAcceptsMetadata>();`` and passing the ```acceptsMetadata ```  to ```AddSupportedRequestFormats(apiDescription.SupportedRequestFormats, hasJsonBody, acceptsMetadata);```

Reply:
@DamianEdwards to answer your question if I do `.Accepts("foo").Accepts("bar", "baz")`, the matcher policy would only allow `bar` and `baz`.  but not `foo`

Reply:
Got it, we only take the last declaration and that makes sense.

Reply:
@DamianEdwards can you take a second look at the APIs description changes. Will be great before we propose them for API reviews.

Reply:
@rafikiassumaniMSFT sure thing, I added some comments. Only nits.

Reply:
@rafikiassumaniMSFT instead of a second lookup, you should just pass the accept metadata you have from the caller.

Reply:
@davidfowl are you referring to the following method ? ```private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string httpMethod, MethodInfo methodInfo)``` ? if yes, that's , I made changes.